src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -338,12 +118,29 @@ void SyncInferRequest::CopyBlobIfNeeded(InferenceEngine::Blob::CPtr src, Inferen
     }
 }
 
-void SyncInferRequest::CopyOutputsIfNeeded() {
-    for (const auto& it : _networkOutputs) {
-        auto& name = it.first;
+void SyncInferRequest::copy_outputs_if_needed() {
+    for (const auto& it : get_outputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        CopyBlobIfNeeded(m_batched_request_wrapper._inferRequestBatched->GetBlob(name), GetBlob(name), false);
+        auto dst_tensor = get_tensor(it);
+        copy_tensor_if_needed(m_batched_request_wrapper->_infer_request_batched->get_tensor(it), dst_tensor, false);
     }
 }
+
+void SyncInferRequest::infer() {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::vector<ov::SoPtr<ov::IVariableState>> SyncInferRequest::query_state() const {
+    auto states = m_batched_request_wrapper->_infer_request_batched->query_state();
+    for (auto&& state : states) {
+        if (!state._so)
+            state._so = m_batched_request_wrapper->_infer_request_batched._so;
+    }
+    return states;
+}
+
+std::vector<ov::ProfilingInfo> SyncInferRequest::get_profiling_info() const {
+    return m_batched_request_wrapper->_infer_request_batched->get_profiling_info();
```

Comment:
same here

Reply:
AUTO_BATCH interface should return _infer_request_batched's profiling info, if fallback to without_batch branch, is it reasonable to get profiling info?

Reply:
SyncInferRequest here correspond to single single infer request, if batch not triggered, we can return the profiling info from single request in batch (I believe should be stored somewhere else already)?

Reply:
ah.. then I see this part should be ok